File: demo/OpenHospital/baseline/frontend/src/types/prescription.ts
export interface PrescriptionRecord {
  id: string
  patient_id: string
  doctor_id: string
  diagnosis: string           // Diagnosis
  treatment_plan: string      // Treatment Plan
  prescribed_tick: number
  status: PrescriptionStatus
  completed_tick?: number
  treatment_result?: TreatmentResult
}

export type PrescriptionStatus = 'pending' | 'completed'

export interface TreatmentResult {
  success: boolean
  score: number
  feedback: string
}

File: demo/OpenHospital/baseline/backend/schemas/prescription.py
"""Prescription schemas."""

from typing import Optional, Literal, Dict, Any
from pydantic import BaseModel


class TreatmentResult(BaseModel):
    """Treatment result."""
    success: bool
    score: float
    feedback: str


class PrescriptionRecord(BaseModel):
    """Prescription record."""
    id: str
    patient_id: str
    doctor_id: str
    diagnosis: str
    treatment_plan: str
    prescribed_tick: int
    status: Literal["pending", "completed"]
    completed_tick: Optional[int] = None
    treatment_result: Optional[TreatmentResult] = None

